src/main/java/io/strimzi/test/container/StrimziKafkaContainer.java

```diff
@@ -33,95 +33,81 @@
-            .withExternalZookeeper(connectString);
-    }
+        LOGGER.info("Injected configuration inside Kafka Container constructor....\n{}", this.kafkaConfigurationMap.toString());
 
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @param useKraft whether the broker should use Kraft
-     * @param additionalKafkaConfiguration additional configuration for Kafka broker
-     * @return instance of @code{StrimziKafkaContainer}
-     */
-    public static StrimziKafkaContainer createWithAdditionalConfiguration(final int brokerId, final boolean useKraft, final Map<String, String> additionalKafkaConfiguration) {
-        return new StrimziKafkaContainer(brokerId, useKraft, additionalKafkaConfiguration);
-    }
+        this.externalZookeeperConnect = builder.externalZookeeperConnect;
 
-    /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @param additionalKafkaConfiguration additional configuration for Kafka broker
-     * @return instance of @code{StrimziKafkaContainer}
-     */
-    public static StrimziKafkaContainer createWithAdditionalConfiguration(final int brokerId, final Map<String, String> additionalKafkaConfiguration) {
-        return createWithAdditionalConfiguration(brokerId, false, additionalKafkaConfiguration);
+        // we need this shared network in case we deploy StrimziKafkaCluster which consist of `StrimziKafkaContainer`
+        // instances and by default each container has its own network, which results in `Unable to resolve address: zookeeper:2181`
+        this.withNetwork(Network.SHARED);
+        // exposing kafka port from the container
+        this.withExposedPorts(KAFKA_PORT);
+        this.withEnv("LOG_DIR", "/tmp");
+
+        this.setDockerImageName("quay.io/strimzi-test-container/test-container:" +
+            this.strimziTestContainerVersion + "-kafka-" +
+            this.kafkaVersion);
+
+        LOGGER.info("=============================================");
+        LOGGER.info(this.getDockerImageName());
+        LOGGER.info("=============================================");
+
+        LOGGER.info("This is inside this object:{}", this::toString);
     }
 
     /**
-     * Static factory method, which creates @code{StrimziKafkaContainer} with empty additional configuration of Kafka broker
-     * @param brokerId broker id
-     * @return instance of StrimziKafkaContainer
+     * Default Kafka port
      */
-    public static StrimziKafkaContainer create(final int brokerId) {
-        return new StrimziKafkaContainer(brokerId, false, Collections.emptyMap());
-    }
+    public static final int KAFKA_PORT = 9092;
+    public static int kafkaDynamicKafkaPort;
+    private static final String DEFAUT_STORAGE_UUID = "xtzWWN4bTjitpL4efd9s6g";
```

Comment:
Is there some value in having this hardcoded? IIRC it was quite easy to generate it.

Reply:
I have added a method in `Utils.randomUuid()` to generate such UUID.... (anyway this code was from Clement... :) )